modules/upgrade/pages/migrate-3x-4x.adoc

```diff
@@ -1,3 +1,445 @@
+The temporary SSH key will be deleted after successful migration.
+Ideally, this is all you will need to do.
+
+Depending on the size of the installation, the actual migration will take up to several hours.
+When successfully finished, you will be notified ("_Migration complete._") and prompted to shut down the source machine, re-configure the network of the target machine to use the same IP address and host name as the original machine and restart it.
+It should now be a fully functional replacement for your previous {productname} 3.2  installation.
+
+The following numbers illustrate the runtime for dumping and importing the database:
+
+----
+14:53:37   Dumping remote database to /var/spacewalk/tmp/susemanager.dmp.gz on target system. Please wait...
+14:58:14   Database successfully dumped. Size is: 506M
+14:58:29   Importing database dump. Please wait...
+15:05:11   Database dump successfully imported.
+----
+
+
+For this example dumping the database takes around five minutes to complete.
+Importing the dump onto the target system will take an additional seven minutes.
+For big installations this can take up to several hours.
+You should also account for the time it takes to copy all the package data to the new machine.
+Depending on your network infrastructure and hardware, this can also take a significant amount of time.
+
+
+
+[[bp.sec.mgr.migration.speedup]]
+== Speeding up the Migration
+
+A complete migration can consume a lot of time.
+This is caused by the amount of data that must be copied.
+Total migration time can be greatly decreased by eliminating the need to copy data prior to performing the migration (for example, channels, packages, auto-install images, and any additional data).
+You can gather all data via {yast} by running:
+
+----
+mgr-setup -r
+----
+
+Executing [command]``mgr-setup -r`` will copy the data from the old server to the new one.
+This command may be run at any time and your current server will remain fully functional.
+When the migration has been initiated only data changed since running [command]``mgr-setup -r`` will need to be transferred.
+This significantly reduces downtime.
+
+On large installations transfering the database (which involves dumping the database onto the source machine and then importing the dump onto the target system) will still take some time.
+During the database transfer no write operations should occur therefore the migration script will shut down any {productname} database services running on the source machine.
+
+
+
+[[bp.sec.mgr.migration.pkg.extern]]
+== Packages on External Storage
+
+
+Some installations may store the package data on external storage (for example, NFS mount on [path]``/var/spacewalk/packages``).
+You do not need to copy this data to the new machine.
+Edit the script located in [path]``/usr/lib/susemanager/bin/mgr-setup`` and remove the respective [command]``rsync`` command (located around line 442).
```

Comment:
```suggestion
After migration has completed, edit the script located in [path]``/usr/lib/susemanager/bin/mgr-setup`` and remove the respective [command]``rsync`` command (it will be around line 442).
```